### ISSUE TITLE
PostgresDatabase and subtypes may produce wrong table comments in presence of stored functions

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#15149]
It appears that this issue never occurred in PostgreSQL, even if it could happen. It has surfaced in CockroachDB now, that we start supporting CockroachDB UDFs (https://github.com/jOOQ/jOOQ/issues/13947). On a fresh database, try this:

```sql
create table t (i int);
comment on table t is 'comment';

select d.objoid, d.classoid, c.relname, d.description
from pg_catalog.pg_class as c
left join pg_catalog.pg_description as d on d.objoid = c.oid
where c.relname = 't';
```

This produces:

```
|objoid|classoid     |relname|description                                                                                    |
|------|-------------|-------|-----------------------------------------------------------------------------------------------|
|108   |4,294,967,106|t      |comment                                                                                        |
|108   |4,294,967,068|t      |Calculates y-intercept of the least-squares-fit linear equation determined by the (X, Y) pairs.|
```

There are now 2 comments generated for this table, where the second comment isn't really a table but a built-in function. Querying the `classoid` values in `pg_class` yields:

```sql
select relname from pg_class where oid in (4294967106, 4294967068);
```

Result:

```
|oid          |relname |
|-------------|--------|
|4,294,967,068|pg_proc |
|4,294,967,106|pg_class|
```

This is also documented here:
https://www.postgresql.org/docs/current/catalog-pg-description.html

So, we should adapt our query to this:

```sql
select d.objoid, d.classoid, c.relname, d.description
from pg_catalog.pg_class as c
left join pg_catalog.pg_description as d on d.objoid = c.oid and d.classoid = 'pg_class'::regclass
where c.relname = 't';
```

Now, we get only the wanted comment:

```
|objoid|classoid     |relname|description|
|------|-------------|-------|-----------|
|108   |4,294,967,106|t      |comment    |
```

**Comment by lukaseder** [comment]
I haven't noticed this in tests, because:

- The tests checked the presence of well known comments, which were correct (by accident) as they were the ones generated (the correnct ones seem to have overridden the incorrect ones)
- The tables that have received wrong comments shouldn't have any comments at all, but there are no tests checking the absence of comments.

**Comment by lukaseder** [comment]
Fixed in versions:

- 3.19.0
- 3.18.5 (https://github.com/jOOQ/jOOQ/issues/15150)
- 3.17.14 (https://github.com/jOOQ/jOOQ/issues/15151)
- 3.16.20 (https://github.com/jOOQ/jOOQ/issues/15152)

**Comment by lukaseder** [comment]
Follow up fix for jOOQ 3.20+:

- #18687
